### DISCUSSION TITLE
Python SDK? anything official or recommended?

### DISCUSSION THREAD

**Discussion by juju4** [openobserve/openobserve#5363] [General]
Found
https://github.com/JustinGuese/python-openobserve (send and search logs)
https://pypi.org/project/openobserve/ (only send logs)

Anything official that can be more confident maintained on the long term? ideally to support integration with jupyter notebooks.

Thanks

**Comment by hengfeiyang** [comment]
We have no SDK yet. what is your requirements for SDK?

**Comment by juju4** [comment]
For now, its mostly about search and I can use the first one above.
In the future, likely anything mapping the API, be it for search, config or logs reporting and more. As you have OpenAPI specs (https://openobserve.ai/docs/api_specs/, json: https://openobserve.ai/docs/api_specs/zinc-specs.json), generating SDK with one of the swagger/openapi generator should be possible. Not sure if one more fit than another but a single reference SDK makes things simpler for users (less code), more consistency and easier debugging for everyone.
* https://github.com/OpenAPITools/openapi-generator (java)
* https://swagger.io/tools/swagger-codegen/ (java)
* https://github.com/fern-api/fern (typescript)

**Reply by prabhatsharma** [reply]
Thanks, @juju4. We know about these tools and some more. We will do it in some time. It's not the highest priority item on our list for now.

**Reply by juju4** [reply]
with python-openobverse or just basic requests query, I always get search error '{"code":500,"message":"Error# [file_list] invalid time range"}'. Start and End time are epoch time in microseconds (also tested nano, mili and seconds just in case with same results).
Possibly related to https://github.com/openobserve/openobserve/issues/2625
Solved by updating module time fields (s/start/start_time/; s/end/end_time/)
But after, while my queries returns HTTP status 200, they don't return any results. same between python-openobserve or curl. 

Only WebUI is providing me expected results
Tested on Version: v0.14.0, Commit Hash: b9435e5992b29d0ab54ce605f912137e11b08c1c and python-openobserve 0.1.3 + python 3.12.7

**Comment by hengfeiyang** [comment]
i didn't try this SDK, but for this error:
```
Error# [file_list] invalid time range"}
```
it means when you run query, the start_time or end_time is empty. The search need with a time range.

**Comment by juju4** [comment]
just adding note that I'm using and enriching https://github.com/JustinGuese/python-openobserve from https://github.com/openobserve/openobserve/issues/1117 but slow merge (https://github.com/JustinGuese/python-openobserve/compare/main...juju4:python-openobserve:devel-all)

top features
* export/import config settings to json/csv/xlsx
* pandas dataframe output for csv, xlsx export but also search post-processing in jupyter notebook for example.
* ongoing work integrations in other libraries
It would be really better to have something official and I agree on maintenance concerns in either case.

**Reply by hengfeiyang** [reply]
@prabhatsharma @juju4 Maybe we can consider let the project under our org but maintain by the users like @juju4
